x/upgrade/abci_test.go

```diff
@@ -5,183 +5,394 @@ import (
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/require"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
+	dbm "github.com/tendermint/tm-db"
+
+	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/simapp"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 	"github.com/cosmos/cosmos-sdk/types/module"
 	"github.com/cosmos/cosmos-sdk/x/gov"
 	"github.com/cosmos/cosmos-sdk/x/upgrade"
-	"github.com/stretchr/testify/suite"
-	abci "github.com/tendermint/tendermint/abci/types"
 )
 
 type TestSuite struct {
-	suite.Suite
-
 	module  module.AppModule
 	keeper  upgrade.Keeper
 	querier sdk.Querier
 	handler gov.Handler
 	ctx     sdk.Context
 }
 
-func (s *TestSuite) SetupTest() {
```

Comment:
why did this change? This was cleaner imo

Reply:
We were using `viper` previously and after your review we understood `viper` should be sticked to `cli` and it should not be used inside `keeper`. The change in code needed a way to set different `skipUpgradeHeights` everytime, so we had to replace the `Suite` with this `setupTest` to expect `skipUpgradeHeights` as a param